python_modules/dagster/dagster/_core/host_representation/code_location.py

```diff
@@ -718,6 +748,102 @@ def has_repository(self, name: str) -> bool:
     def get_repositories(self) -> Mapping[str, ExternalRepository]:
         return self.external_repositories
 
+    def launch_resource_verification(
+        self,
+        origin: ExternalRepositoryOrigin,
+        instance: DagsterInstance,
+        resource_name: str,
+    ) -> ResourceVerificationResult:
+        from dagster._config.structured_config.resource_verification import (
+            VerificationResult,
+            VerificationStatus,
+            _ResourceVerificationContext,
+        )
+        from dagster._core.definitions.run_request import InstigatorType
+        from dagster._core.scheduler.instigation import TickData, TickStatus
+        from dagster._utils.error import (
+            serializable_error_info_from_exc_info,
+        )
+        from dagster._utils.log import get_dagster_logger
+
+        timestamp = time.time()
+        tick = instance.create_tick(
+            TickData(
+                instigator_origin_id=origin.get_id(),
+                instigator_name=f"resource_verification_{resource_name}",
+                instigator_type=InstigatorType.VERIFICATION,
+                status=TickStatus.STARTED,
+                timestamp=timestamp,
+                selector_id=resource_name,
+            )
+        )
+        try:
+            external_resource = self.get_repositories()[
+                origin.repository_name
+            ].get_external_resource(resource_name)
+        except KeyError:
+            return ResourceVerificationResult(
+                VerificationResult.failure(f"Resource {resource_name} not found"), None
+            )
+        logger = get_dagster_logger()
+        tick_retention_settings = instance.get_tick_retention_settings(InstigatorType.VERIFICATION)
+
+        with _ResourceVerificationContext(
+            external_resource, tick, instance, logger, tick_retention_settings
+        ) as tick_context:
+            try:
+                result = self.client.resource_verification(
```

Comment:
This uses a direct GRPC call vs a yield like the other instigator invocations, not sure if that needs to be changed.